src/views/Todo/Todo.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from 'react';
+import TodoList from '../../components/TodoList/TodoList';
+import { fetchTodos, createTodo, toggleCompleted, deleteChecked } from '../../services/todos';
+
+export default function Todo() {
+  const [task, setTask] = useState('');
+  const [todoList, setTodoList] = useState([]);
+  const [message, setMessage] = useState('');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const data = await fetchTodos();
+      setTodoList(data);
+      if (data.length < 1) setMessage('no tasks at this time');
+    };
+    fetchData();
+  }, []);
+
+  const addNewTask = async (e) => {
+    try {
+      e.preventDefault();
+      const resp = await createTodo(task);
+      setTask('');
+      setMessage('');
+      setTodoList((prevState) => [resp[0], ...prevState]);
+    } catch {
+      setMessage('tasks must be more than 3 characters in length');
+    }
+  };
+
+  const handleCheck = async (task) => {
+    await toggleCompleted(task.id, !task.is_complete);
+
+    setTodoList((prevState) =>
+      prevState.map((item) =>
+        item.id === task.id ? { ...task, is_complete: !task.is_complete } : item
+      )
+    );
+  };
+
+  const handleClear = async () => {
+    if (confirm('do you want to erase all completed tasks?')) {
+      await deleteChecked();
+
+      const resp = await fetchTodos();
```

Comment:
I might use the same pattern here that you used for the toggle / complete for consistency. We could filter state and set the message based on the array returned from our filter.